Meshtastic/MeshtasticAppDelegate.swift

```diff
@@ -38,14 +38,72 @@ class MeshtasticAppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificat
 	) {
 		completionHandler([.list, .banner, .sound])
 	}
-
+	
 	// This method is called when a user clicks on the notification
 	func userNotificationCenter(
 		_ center: UNUserNotificationCenter,
 		didReceive response: UNNotificationResponse,
 		withCompletionHandler completionHandler: @escaping () -> Void
 	) {
 		let userInfo = response.notification.request.content.userInfo
+		
+		switch response.actionIdentifier {
+		case UNNotificationDefaultActionIdentifier:
+			break
+			
+		case "messageNotification.thumbsUpAction":
+			if let channel = userInfo["channel"] as? Int32,
+			   let replyID = userInfo["messageId"] as? Int64 {
+				let tapbackResponse = !BLEManager.shared.sendMessage(
+					message: Tapbacks.thumbsUp.emojiString,
+					toUserNum: userInfo["userNum"] as? Int64 ?? 0,
+					channel: channel,
+					isEmoji: true,
+					replyID: replyID
+				)
+				Logger.services.info("Tapback response sent")
+			} else {
+				Logger.services.error("Failed to retrieve channel or messageId from userInfo")
+			}
+			break
+			
+		case "messageNotification.thumbsDownAction":
+			if let channel = userInfo["channel"] as? Int32,
+			   let replyID = userInfo["messageId"] as? Int64 {
+				let tapbackResponse = !BLEManager.shared.sendMessage(
+					message: Tapbacks.thumbsDown.emojiString,
+					toUserNum: userInfo["userNum"] as? Int64 ?? 0,
+					channel: channel,
+					isEmoji: true,
+					replyID: replyID
+				)
+				Logger.services.info("Tapback response sent")
+			} else {
+				Logger.services.error("Failed to retrieve channel or messageId from userInfo")
+			}
+			break
+			
+		case "messageNotification.replyInputAction":
+			if let userInput = (response as? UNTextInputNotificationResponse)?.userText,
+			   let channel = userInfo["channel"] as? Int32,
+			   let replyID = userInfo["messageId"] as? Int64 {
+				let tapbackResponse = !BLEManager.shared.sendMessage(
```

Comment:
Two questions here:
* It looks like for each of these the variable `tapbackResponse` is never used. Can they be removed?
* Do we need to consider any edge-cases that could cause this to crash? (ie: BLE is disconnected)

Reply:
sendMessage returns a boolean if it fails or goes through. For logging purposes we could use it.